### PR TITLE
Setup integration tests for dependency injection.

### PR DESCRIPTION
**Pull request by gdbelvin**
This PR removes the `/integration` directory and creates 
- `core/integration` - integration test functions and `*Env`.
- `impl/integration` - integration test runner and concrete implementation of `*Env`

This allows other implementations to also run the integration tests with their own `*Env` implementations.

### CODE REVIEW

File: integration/env.go [hunk 1]
```diff
@@ -125,24 +112,24 @@ func NewEnv(t *testing.T) *Env {
...
-		t.Fatalf("Failed to load vrf pubkey: %v", err)
+		return nil, fmt.Errorf("env: Failed to load vrf pubkey: %v", err)
 	}
 
 	// Common data structures.
-	mutations, err := mutationstorage.New(sqldb)
+	mutations, err := mutationstorage.New(db)
 	if err != nil {
-		log.Fatalf("Failed to create mutations object: %v", err)
+		log.Fatalf("env: Failed to create mutations object: %v", err)
```
Comment (phad) [inline integration/env.go]: Could this one be Errorf() like the previous ones you changed?

File: integration/main_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"testing"
+
+	"github.com/google/trillian/storage/testdb"
+)
+
+// TestIntegration runs all KeyTransparency integration tests.
+func TestIntegration(t *testing.T) {
+	// We can only run the integration tests if there is a MySQL instance available.
+	if provider := testdb.Default(); !provider.IsMySQL() {
+		t.Skipf("Skipping map integration test, SQL driver is %v", provider.Driver)
```
Comment (phad) [inline integration/main_test.go]: s/map/KT/  ?

File: integration/env.go (hunk 1 above)
Comment (gdbelvin) [reply integration/env.go]: yep, missed that one.

File: integration/main_test.go (hunk 2 above)
Comment (gdbelvin) [reply integration/main_test.go]: yep

**Pull request merged**